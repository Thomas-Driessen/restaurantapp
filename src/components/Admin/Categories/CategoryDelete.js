import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import RemoveIcon from '@material-ui/icons/Remove';
import { IconButton } from '@material-ui/core';

const Product = (props) => {
  const [del, setDelete] = React.useState(false);

  const handleDeleteOpen = () => {
    setDelete(true);
  };

  const handleDeleteClose = () => {
    setDelete(false);
  };

  const deleteCategory = () => {
    fetch(`${process.env.REACT_APP_API_URL}/api/category/${props.categoryType}`, {
      method: "DELETE",
      mode: "cors",
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(props.category)
    }).then(response => response.json())
    alert("Your category has been deleted");
    setDelete(false);
  }

  return (
    <div>
      <IconButton
        size="medium"
        color="primary"
        onClick={handleDeleteOpen}
      >
        <RemoveIcon />
      </IconButton>

      <Dialog open={del} onClose={handleDeleteClose} aria-labelledby="product-title">
        <DialogTitle id="product-title">
          {props.category.categoryName}
        </DialogTitle>
        <DialogContent>
          <DialogContentText>
            Are you sure you want to delete this Category?
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button size="medium" color="primary" onClick={deleteCategory}>
            Yes
          </Button>
          <Button size="medium" color="primary" onClick={handleDeleteClose}>
            No
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}
export default Product