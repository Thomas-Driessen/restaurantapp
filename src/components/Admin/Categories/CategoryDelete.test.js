import React from 'react';
import { render } from '@testing-library/react';
import CategoryDelete from './CategoryDelete';
import renderer from 'react-test-renderer';

const category = {
  categoryName: "Cold"
};

it('CategoryDelete renders correctly', () => {
  const tree = renderer
    .create(<CategoryDelete category={category} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render CategoryDelete', () => {
  const { getByText } = render(<CategoryDelete category={category} />);
  const linkElement = getByText(/Delete/i);
  expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
  global.gc && global.gc()
})