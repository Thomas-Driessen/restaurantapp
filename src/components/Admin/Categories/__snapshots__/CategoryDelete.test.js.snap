// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryDelete renders correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    target="_blank"
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Delete
      </span>
    </span>
  </button>
</div>
`;
